calibFuncExample.py

```diff
@@ -24,56 +24,70 @@
-    # setting base parameters could be needed for non standard functions. Consult the TF1 reference.
-    fucalib.Draw()
-    fucalib.Write()
-
-    nparFunc = fucalib.GetNpar()
-
-    for par in range(nparFunc):
-      histoParaName = baseCalibParaName + "_" + str(par) # same name for all sensors but in different folder
-      hpara = TH2F(histoParaName, "", cols, 0, cols, rows, 0, rows) # creating histogram for parameters
-
-      weight = par + 1 # get your parameters from a dedicated calibration probably
-      for c in range(cols):
-        for r in range(rows):
-          hpara.Fill(c, r, 0)
-      hpara.Write()
-
-  foutfile.Close()
+    # Parse the command line
+    args = parse_args()
+
+    # parameters to provide
+    cols = 512
+    rows = 512
+
+    sensorIDList = [22]
+
+    baseCalibFuncName = "calibFunc"
+    baseCalibParaName = "para"
+
+    gainCalibrationFileName = args.output
+    dut = args.dut
+
+    # generating file structure
+    foutfile = TFile(gainCalibrationFileName, "RECREATE")
+
+    for sensorID in sensorIDList:
+        foutfile.cd()
+        detDir = foutfile.mkdir("d" + str(sensorID))  # creating folder for sensor
+        detDir.cd()
+        funcName = baseCalibFuncName  # function name is the same for all sensors in the file, differentiated by the folders
+        fucalib = TF1(funcName, "10.1*(-[1] + [0] * [3] + x + sqrt([1]**2 + 4 * [0]* [2] + 2 *[0]* [1] *[3] + [0]**2 * [3]**2 - 2 *[1] *x - 2 *[0] *[3] *x + x**2))/(2 *[0])", 0.0, 128.0)  # creating calibration function, use whatever you need as function and range
+        # ucalib = TF1(funcName, "1+0*(-[1] + [0] * [3] + x + sqrt([1]**2 + 4 * [0]* [2] + 2 *[0]* [1] *[3] + [0]**2 * [3]**2 - 2 *[1] *x - 2 *[0] *[3] *x + x**2))/(2 *[0])", 0.0, 128.0)  # creating calibration function, use whatever you need as function and range
+
+        # setting base parameters could be needed for non standard functions. Consult the TF1 reference.
+        fucalib.SetParameters(0, 0.134)  # just set some values from W14R12 NF
+        fucalib.SetParameters(1, 2.6)  # just set some values from W14R12 NF
+        fucalib.SetParameters(2, 1.9e02)  # just set some values from W14R12 NF
+        fucalib.SetParameters(3, 26)  # just set some values from W14R12 NF
+
+        fucalib.Draw()
+        fucalib.Write()
+
+        nparFunc = fucalib.GetNpar()
+
+        for par in range(nparFunc):
+            histoParaName = baseCalibParaName + "_" + str(par)  # same name for all sensors but in different folder
+            hpara = TH2F(histoParaName, "", cols, 0, cols, rows, 0, rows)  # creating histogram for parameters
+
+            weight = par + 1  # get your parameters from a dedicated calibration probably
+            for c in range(cols):
+                for r in range(rows):
+                    arr = df[str(c)][r][1:-1].split(',')
+                    hpara.Fill(c, r, float(arr[par]))
```

Comment:
this is a horrible way of reading your dataframe back. 

you can get accces to indivual data fields from integers using `df.iloc[c,r]` syntax. 
No need to convert `c` to str or have intermediate array `arr` or the other funny stuff. 
I would recommend you to debug by adding a `print` statement. 

And i would recommend to have a jupyter notebook to read back the of all these TH2F from 
the calib root file and print col row value. 

Just to make sure your root file are really correct and workable. 


i would definitely